src/js/components/FileInput/FileInput.js

```diff
@@ -461,29 +462,42 @@ const FileInput = forwardRef(
                 justify="between"
                 direction="row"
                 align="center"
+                style={{ zIndex: 0 }} // make the tip appear on file name hover
               >
                 {renderFile ? (
                   renderFile(file)
                 ) : (
-                  <Box
-                    {...theme.fileInput.label}
-                    gap="xsmall"
-                    align="center"
-                    direction="row"
+                  <Tip
+                    content={file.name}
+                    dropProps={{
+                      align: { top: 'bottom' },
+                      width: 'small',
+                      round: 'xsmall',
+                      background: { ...theme.tip.drop },
+                    }}
                   >
-                    {((maxSize && file.size > maxSize) ||
-                      (max && index >= max)) && <CircleAlert />}
-                    <Label
-                      weight={
-                        theme.global.input.weight ||
-                        theme.global.input.font.weight
-                      }
-                      truncate
-                      {...passThemeFlag}
+                    <Box
+                      {...theme.fileInput.label}
+                      gap="xsmall"
+                      align="center"
+                      direction="row"
+                      tabIndex={0} // make name focusable with keyboard
```

Comment:
in the case you mentioned 
`keyboard only users who don't use a screenreader and hence they won't have access to the full name when they navigate with Keyboard only
`

can we only set this `tabIndex={0} // make name focusable with keyboard` for when the text is truncated